grpcutils/dial.go

```diff
@@ -59,8 +62,13 @@ func (o DialOpts) Dial(ctx context.Context) (*grpc.ClientConn, error) {
 		opts = append(opts, grpc.WithAuthority(o.Authority))
 	}
 	opts = append(opts, o.ExtraOptions...)
-	ctx, cancel := context.WithTimeout(ctx, time.Minute)
-	defer cancel()
+
+	ctx, cancelFn := context.WithTimeout(ctx, (time.Minute * 1))
+	if o.Timeout > 0 {
+		ctx, cancelFn = context.WithTimeout(ctx, o.Timeout)
+	}
+	defer cancelFn()
```

Comment:
why create a cancel and then only cancel it at the end of the function? why not just create a new context and let it go out of scope?

also i think this is slightly easier to read like so

```golang
        timeout := time.Minute
	if !o.Timeout.IsZero() {
		timeout = o.Timeout
	}
       ctx = context.WithTimeout(timeout)
```

Reply:
I think we agreed on this keeping the same behavior but making the timeout value optional.

Reply:
Could you suggest the changes? I don't see `IsZero` methood on time.Duration @kdorosh

Reply:
updated, confused `time.Time` with `time.Duration` but we can still just compare it to zero

Reply:
Updated